### PR TITLE
added preprod env variable for respondent accounts url

### PR DESCRIPTION
**Pull request by MIslam92**
### What is the context of this PR?
In relation to : 
https://trello.com/c/7VACxFOq/1016-clear-actions-to-exit-or-continue-a-survey-s
See: https://github.com/ONSdigital/eq-survey-runner/pull/1274
Having to add a variable for preprod for the respondent accounts url for when terraform is ran and deployed.

### How to review
Get Jon to review

### CODE REVIEW

File: survey-runner.tf [hunk 1]
```diff
@@ -37,7 +37,8 @@ module "survey-runner-on-beanstalk" {
   dns_zone_name = "${var.dns_zone_name}"
   deployment_policy = "${var.eb_deployment_policy}"
   rolling_update_enabled = "${var.eb_rolling_update_enabled}"
-  secrets_file_name       = "${var.survey_runner_secrets_file_name}"
+  secrets_file_name      = "${var.survey_runner_secrets_file_name}"
+  respondent_account_url = "${var.respondent_accounts_url}"
```
Comment (jonnyshaw89) [inline survey-runner.tf]: Looks like there is a name mismatch? `respondent_account(s)_url`

Comment (jonnyshaw89) [reply survey-runner.tf]: Where is this Var coming from?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: survey-runner-application/global_vars.tf [hunk 2]
```diff
@@ -125,6 +125,11 @@ variable "secrets_file_name" {
   default     = "secrets.yml"
 }
 
+variable "respondent_accounts_url" {
```
Comment (jonnyshaw89) [inline survey-runner-application/global_vars.tf]: Where are you using this?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: survey-runner.tf (hunk 1 above)
Comment (MIslam92) [reply survey-runner.tf]: So in the other PR: https://github.com/ONSdigital/eq-survey-runner/pull/1274/commits/2b71a9dd7d7f9c8c38b3220abc71b448bc4bc63f
these changes where recently made and for preprod it will be a different url

Comment (jonnyshaw89) [reply survey-runner.tf]: So you need to pass `respondent_account_url` into an environment variable in Beanstalk as `RESPONDENT_ACCOUNT_URL`

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: survey-runner.tf (hunk 1 above)
Comment (jonnyshaw89) [reply survey-runner.tf]: Like these... https://github.com/ONSdigital/eq-terraform/blob/master/survey-runner-application/survey_runner.tf#L220

File: survey-runner-application/global_vars.tf [hunk 3]
```diff
@@ -125,8 +125,8 @@ variable "secrets_file_name" {
   default     = "secrets.yml"
 }
 
-variable "respondent_accounts_url" {
-  description = "The url for the respondent log in that will be used to navigate n preprod"
+variable "respondent_account_url" {
+  description = "The url for the respondent log in that will be used to navigate to preprod"
   default     = "https://surveypreprod.ons.gov.uk/"
```
Comment (jonnyshaw89) [inline survey-runner-application/global_vars.tf]: Do you want to set this default to real value?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

**Comment by jonnyshaw89** [comment]
@MIslam92 you will also need to modify the ECS survey runner with this variable

**Comment by shahi645** [comment]
@MIslam92  getting errors running `terraform plan`

File: survey-runner.tf [hunk 4]
```diff
@@ -37,7 +37,8 @@ module "survey-runner-on-beanstalk" {
   dns_zone_name = "${var.dns_zone_name}"
   deployment_policy = "${var.eb_deployment_policy}"
   rolling_update_enabled = "${var.eb_rolling_update_enabled}"
-  secrets_file_name       = "${var.survey_runner_secrets_file_name}"
+  secrets_file_name      = "${var.survey_runner_secrets_file_name}"
+  respondent_account_url = "${var.respondent_account_url}"
```
Comment (jonnyshaw89) [inline survey-runner.tf]: where does `"${var.respondent_account_url}"` come from?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: survey-runner.tf (hunk 4 above)
Comment (MIslam92) [reply survey-runner.tf]: @jonnyshaw89 From the global.vars.tf? I thought

Comment (jonnyshaw89) [reply survey-runner.tf]: The `global_vars.tf` that you have modified is in a separate module (folder). You will need to add this to the developer defaults. if you try running this branch its clear that it is missing

**Comment by shahi645** [comment]
@MIslam92 still getting this error:
`* module root: module survey-runner-on-ecs: respondent_account_url is not a valid parameter`

**Comment by jonnyshaw89** [comment]
@shahi645 you need to point the module at the correct branch 
`"github.com/ONSdigital/eq-survey-runner-deploy?ref=eq-1016-add-preprod-variable-to-deploy"`

File: survey-runner.tf [hunk 5]
```diff
@@ -38,7 +38,6 @@ module "survey-runner-on-beanstalk" {
   deployment_policy = "${var.eb_deployment_policy}"
   rolling_update_enabled = "${var.eb_rolling_update_enabled}"
   secrets_file_name      = "${var.survey_runner_secrets_file_name}"
-  respondent_account_url = "${var.respondent_account_url}"
```
Comment (jonnyshaw89) [inline survey-runner.tf]: Why did you remove this?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: survey-runner.tf [hunk 6]
```diff
@@ -58,7 +59,7 @@ module "eq-ecs" {
 }
 
 module "survey-runner-on-ecs" {
-  source                  = "github.com/ONSdigital/eq-survey-runner-deploy"
+  source                  = "github.com/ONSdigital/eq-survey-runner-deploy?ref=eq-1016-add-preprod-variable-to-deploy"
```
Comment (jonnyshaw89) [inline survey-runner.tf]: You don't need this branch ref now its merged.

Comment (jonnyshaw89) [reply survey-runner.tf]: @MIslam92 if you remove this i can tick it.

Comment (MIslam92) [reply survey-runner.tf]: ?? I've removed this ??

**Review by jonnyshaw89 — APPROVED** [review]

File: developer_defaults.tf [hunk 7]
```diff
@@ -167,6 +167,11 @@ variable "survey_runner_min_tasks" {
   default     = "1"
 }
 
+variable "respondent_account_url" {
+  description = "The url for the respondent log in that will be used to navigate to preprod"
```
Comment (ajmaddaford) [inline developer_defaults.tf]: This description isn't quite right - the `that will be used to navigate to preprod` bit should be dropped.

File: survey-runner-application/global_vars.tf [hunk 8]
```diff
@@ -125,6 +125,11 @@ variable "secrets_file_name" {
   default     = "secrets.yml"
 }
 
+variable "respondent_account_url" {
+  description = "The url for the respondent log in that will be used to navigate to preprod"
```
Comment (ajmaddaford) [inline survey-runner-application/global_vars.tf]: Same as above

File: survey-runner.tf [hunk 9]
```diff
@@ -37,7 +37,8 @@ module "survey-runner-on-beanstalk" {
   dns_zone_name = "${var.dns_zone_name}"
   deployment_policy = "${var.eb_deployment_policy}"
   rolling_update_enabled = "${var.eb_rolling_update_enabled}"
-  secrets_file_name       = "${var.survey_runner_secrets_file_name}"
+  secrets_file_name      = "${var.survey_runner_secrets_file_name}"
+  respondent_account_url  = "${var.respondent_account_url}"
```
Comment (ajmaddaford) [inline survey-runner.tf]: Spacing before `=`

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

**Review by ajmaddaford — APPROVED** [review]

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]
I think you are missing passing this new variable into the survey runner on ecs?

**Review by jonnyshaw89 — APPROVED** [review]

**Comment by jonnyshaw89** [comment]
@MIslam92 can you squash this again?
